python/ray/data/tests/test_stats.py

```diff
@@ -1488,6 +1488,55 @@ def test_spilled_stats(shutdown_only):
     assert ds._plan.stats().dataset_bytes_spilled == 0
 
 
+def test_stats_actor_metrics():
+    ray.init(object_store_memory=100e6, num_gpus=1)
+    with patch(
+        "ray.data._internal.execution.streaming_executor._update_stats_actor_metrics"
+    ) as update_fn:
+        ds = ray.data.range(1000 * 80 * 80 * 4).map_batches(lambda x: x).materialize()
+
+    last_stats = update_fn.call_args_list[-1].args[0]
```

Comment:
I'm surprised that `update_fn` is still accessible after the `with patch() as update_fn` block.

Instead of patching the `_update_stats_actor_metrics` function, I think we can also query the latest stats from the stats actor. So we the test can cover more code.

Reply:
Is there a way to get the value from a metrics object?

Reply:
The value is not exposed to Python. Never mind, the current way is fine as well.